File: Java/Car.java
// class Car {
//   Integer id;
//   String license;
//   // heredar de Account.java
//   Account driver;
//   Integer passegenger;

//   // Declarando un Método Constructor en Java 
//   public Car(String license, Account driver){
//     this.license = license;
//     this.driver = driver;
// }

//   // metodo para llamar los datos
//   void printDataCar() {
//       // System.out.println("License: " + license + " Driver: " + driver);
//       System.out.println("License: " + license + " Name Driver: " + driver.name);
//     }
//   }
class Car {
  Integer id;
  String license;
  Account driver;
  private Integer passegenger;

  public Car(String license, Account driver){
      this.license = license;
      this.driver = driver;
  }

  void printDataCar() {
      if(passegenger != null){
          System.out.println("License: " + license + " Name Driver: " + driver.name + " Passengers: " + passegenger);
      }
      
  }


  public Integer getPassenger(){
      return passegenger;
  }

  public void setPassenger(Integer passenger) {
      if(passenger == 4){
          this.passegenger = passenger;
      }else{
          System.out.println("Necesitas asignar 4 pasajeros");
      }
  }
  
}

File: JS/Car.js
// Declarando un Método Constructor en Java y JavaScript
// function Car(license, driver) {
//   this.id;
//   this.license = license;
//   this.driver = driver;
//   this.passenger;
// }

// Car.prototype.printDataCar = function () {
//   console.log(this.driver)
//   console.log(this.driver.name)
//   console.log(this.driver.document)
// }
// /************************************** */ECMAScript6 //////////////////////////*
class Car {
  constructor(license, driver){
    this.id;
    this.license = license;
    this.driver = driver;
    this.passanger;
  }
  printDataCar (){
    console.log(this.driver)
    console.log(this.driver.name)
    console.log(this.driver.document)
  }
}